sql/src/main/java/io/crate/executor/transport/task/UpdateByIdTask.java

```diff
@@ -0,0 +1,155 @@
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+
+import static io.crate.executor.transport.task.DeleteByIdTask.getShardId;
+
+public class UpdateByIdTask extends JobTask {
+
+    private final ClusterService clusterService;
+    private final Functions functions;
+    private final TransportShardUpsertAction shardUpsertAction;
+    private final UpdateById updateById;
+    private final Function<Boolean, ShardUpsertRequest.Builder> createBuilder;
+    private final Assignments assignments;
+
+    public UpdateByIdTask(ClusterService clusterService,
+                          Functions functions,
+                          TransportShardUpsertAction shardUpsertAction,
+                          UpdateById updateById) {
+        super(updateById.jobId());
+        this.clusterService = clusterService;
+        this.functions = functions;
+        this.shardUpsertAction = shardUpsertAction;
+        this.updateById = updateById;
+        assignments = Assignments.convert(updateById.assignmentByTargetCol());
+        createBuilder = (continueOnError) ->
+            new ShardUpsertRequest.Builder(
+                ShardingUpsertExecutor.BULK_REQUEST_TIMEOUT_SETTING.setting().get(clusterService.state().metaData().settings()),
+                false,
+                continueOnError,
+                assignments.targetNames(),
+                null, // missing assignments are for INSERT .. ON DUPLICATE KEY UPDATE
+                updateById.jobId(),
+                false
+            );
+    }
+
+    @Override
+    public void execute(RowConsumer consumer, Row parameters) {
+        HashMap<ShardId, ShardUpsertRequest> requestsByShard = new HashMap<>();
+        ShardUpsertRequest.Builder requestBuilder = createBuilder.apply(updateById.docKeys().size() > 1);
+        addRequests(0, parameters, requestsByShard, requestBuilder);
+        MultiActionListener<ShardResponse, long[], ? super Row> listener = new MultiActionListener<>(
+            requestsByShard.size(),
+            () -> new long[]{0},
+            DeleteByIdTask::updateRowCountOrFail,
+            rowCount -> new Row1(rowCount[0]),
+            new OneRowActionListener<>(consumer, Function.identity())
+        );
+        for (ShardUpsertRequest request : requestsByShard.values()) {
+            shardUpsertAction.execute(request, listener);
+        }
+    }
+
+    @Override
+    public List<CompletableFuture<Long>> executeBulk(List<Row> bulkParams) {
+        HashMap<ShardId, ShardUpsertRequest> requestsByShard = new HashMap<>();
+        ShardUpsertRequest.Builder requestBuilder = createBuilder.apply(true);
+        int location = 0;
+        // idx=location; value=resultIdx
+        IntArrayList bulkIndices = new IntArrayList(bulkParams.size() * updateById.docKeys().size());
+        for (int resultIdx = 0; resultIdx < bulkParams.size(); resultIdx++) {
```

Comment:
would it makes sense to extract this into a shared method as this looks identically to whats happening at the DeleteByIdTask?